library/core/src/androidMain/kotlin/com/attafitamim/krop/core/crop/ImageCropper.android.kt

```diff
@@ -41,14 +47,35 @@ suspend fun ImageCropper.crop(
     crop(maxResultSize) { cached?.toImageSrc(context) }
 }
 
-const val CacheDir = "krop_cache"
+/**
+ * Initiates a new crop session, cancelling the current one, if any.
+ * Suspends until a result is available (cancellation, error, success) and returns it.
+ * The resulting image will be scaled down to fit [maxResultSize] if provided.
+ * [dataSource] will be used as a source.
+ * Set [cacheBeforeUse] to false if you're certain that reopening it multiple times won't be a problem,
+ * true otherwise.
+ */
+suspend fun ImageCropper.crop(
+    dataSource: ImageStream,
+    context: Context,
+    maxResultSize: IntSize? = DefaultMaxCropSize,
+    cacheBeforeUse: Boolean = true
+): CropResult = cacheStream(enabled = cacheBeforeUse, dataSource, context) { cached ->
+    crop(maxResultSize) {
+        cached?.let {
+            ImageStreamSrc(cached)
+        }
+    }
+}
 
 suspend fun <R> cacheUri(
-    enabled: Boolean, uri: Uri, context: Context,
+    enabled: Boolean,
+    uri: Uri,
+    context: Context,
     block: suspend (Uri?) -> R
 ): R {
     if (!enabled) return block(uri)
-    val dst = context.cacheDir.resolve("$CacheDir/${UUID.randomUUID()}")
+    val dst = context.cacheDir.resolve("$CropCacheDir/${UUID.randomUUID()}")
```

Comment:
Is this what the boolean parameter that I mentioned above does? Why do we need this?